pkg/watch/dockerignore.go

```diff
@@ -61,13 +63,28 @@ func (i dockerPathMatcher) MatchesEntireDir(f string) (bool, error) {
 	return true, nil
 }
 
-func LoadDockerIgnore(repoRoot string) (*dockerPathMatcher, error) {
+func LoadDockerIgnore(build *types.BuildConfig) (*dockerPathMatcher, error) {
+	repoRoot := build.Context
 	absRoot, err := filepath.Abs(repoRoot)
 	if err != nil {
 		return nil, err
 	}
 
-	patterns, err := readDockerignorePatterns(absRoot)
+	// first try Dockerfile-specific ignore-file
+	f, err := os.Open(filepath.Join(repoRoot, build.Dockerfile+".dockerignore"))
```

Comment:
Is `build.Dockerfile` always just the file name, or can it be an absolute path? (in that case, no need to join if with the repo-root)

Reply:
I'm actually running some tests as I wondered watch do abspath resolution, while compose-go should already have done

Reply:
`Dockerfile` is always just the file name within the context